src/dds.imageio/ddsinput.cpp

```diff
@@ -636,7 +717,18 @@ DDSInput::seek_subimage(int subimage, int miplevel)
     return true;
 }
 
-
+static uint32_t
+bit_convert(uint32_t value, uint32_t in_bits, uint32_t out_bits)
```

Comment:
In fmath.h, we have `bit_range_convert()` (both a templated version for when you know the in and out bit depths, and a function version for when you don't).

Can you look that over and be sure it does the right/same thing as this, and if so, let's use that version. (Or if you don't think the one in fmath does the right thing, please do let me know.)

Reply:
Ah cool! Yeah that does exactly what is needed here, updated the code to use it.